pkg/config/resource.go

```diff
@@ -212,6 +212,12 @@ type Resource struct {
 	// databases.
 	UseAsync bool
 
+	// AllowDestroy controls whether the `prevent_destroy: true` stanza
+	// is included in the Terraform configuration for the resource.
+	// For certain resources like Azure's PostgreSQLConfiguration,
+	// we would like to allow replacement.
+	AllowDestroy bool
```

Comment:
```suggestion
	// AllowRecreation controls whether the changes that require resource
	// re-creation, i.e. immutable fields, should be allowed, which is against
	// Crossplane Resource Model.
	//
	// This should be enabled only in cases where there is no point in lifecycle
	// of the resource where the API returns NotFound. For example, the
	// configurations that Azure PostgreSQLConfiguration resource changes
	// have defaults, hence they always exist and when you create a new
	// PostgreSQLConfiguration, it's always an importing process of the default
	// so the operation would have to be a re-creation since all parameters are
	// immutable. Otherwise, controller imports instead of creation since the
	// resource exists and ends up in dead-lock state since re-creation is no
	// allowed.
	//
	// For other cases, enabling this is strictly against Crossplane Resource Model
	// where the deletion of the external resource is issued if and only if the custom
	// resource in Kubernetes API is deleted. Use this parameter with high caution.
	AllowRecreation bool
```